packages/angular/cli/src/command-builder/command-module.ts

```diff
@@ -160,6 +167,10 @@ export abstract class CommandModule<T extends {} = {}> implements CommandModuleI
         throw e;
       }
     } finally {
+      // Flush one last time.
+      clearInterval(analyticsFlushInterval);
+      await analyticsFlushPromise.then(() => analytics.flush());
```

Comment:
This looks a lot like an Observable pattern. I'm wondering if we should consider using that or maybe an async generator to manage the `setInterval()` / `clearInterval()` part in a more ergonomic way?

Reply:
I updated this but I avoid to use RxJs as it's not a dependency and in this case while it was possible to use  generators it does make it more complex. 

Therefore I went with a simple method that returns a method which can be called to run the teardown logic.